pkg/services/sqlstore/datasource_test.go

```diff
@@ -214,4 +215,27 @@ func TestDataAccess(t *testing.T) {
 
 		require.Equal(t, 0, len(query.Result))
 	})
+
+	t.Run("GetDataSource", func(t *testing.T) {
+		t.Run("Number of data sources returned limited to 6 per organization", func(t *testing.T) {
+			InitTestDB(t)
+			datasourceLimit := 6
+			for i := 0; i < datasourceLimit+1; i++ {
+				err := AddDataSource(&models.AddDataSourceCommand{
+					OrgId:    10,
+					Name:     "laban" + strconv.Itoa(i),
+					Type:     models.DS_GRAPHITE,
+					Access:   models.DS_ACCESS_DIRECT,
+					Url:      "http://test",
+					Database: "site",
+					ReadOnly: true,
+				})
+				require.NoError(t, err)
+			}
+			query := models.GetDataSourcesQuery{OrgId: 10, DataSourceLimit: &datasourceLimit}
+			err := GetDataSources(&query)
+			require.NoError(t, err)
+			require.Equal(t, datasourceLimit, len(query.Result))
```

Comment:
Would it make sense to do a validation that the result is larger than the data source limit if no data source limit is set?

i.e. adding this to the bottom of the test

```go
query := models.GetDataSourcesQuery{OrgId: 10}
err := getDataSources(&query)
require.NoError(t, err)
assert.Len(t, query.Result, datasourceLimit+1)
```